Fix #20088 Failed to run install_fedora_deps()!!! with Vagrant

arulrajnet opened the pull request:
- Saltstack script https://bootstrap.saltstack.com using http://copr.fedoraproject.org/coprs/saltstack/salt/repo/fedora-21/saltstack-salt-fedora-21.repo at line no 2988
- Curl doesn't follow redirect
- `BS_CURL_ARGS` variable to override cURL arguments
- added BS_CURL_ARGS="-ksSL" in provision-master.sh and provission-minion.sh

arulrajnet committed:
Fix #20088 Failed to run install_fedora_deps()!!! with Vagrant

- Saltstack script https://bootstrap.saltstack.com using http://copr.fedoraproject.org/coprs/saltstack/salt/repo/fedora-21/saltstack-salt-fedora-21.repo at line no 2988
- Curl doesn't follow redirect
- `BS_CURL_ARGS` variable to override cURL arguments
- added BS_CURL_ARGS="-L" in provision-master.sh and provission-minion.sh
- Comment added and mentioned this fix as temporary fix

Resolving comments of @marun for #20088
- Comment added and mentioned this fix as temporary fix
- instead of "-ksSL" used "-L" only. Because others are not needed.

fabioy commented:
@k8s-bot ok to test

marun commented:
@arulrajnet Looks like https://github.com/saltstack/salt-bootstrap/commit/f8715cb97fe18349567ef03bf37c130e53a27b49 switched to using the correct url and https://github.com/saltstack/salt-bootstrap/commit/4d4de1ec7488dc03afc3f1af980ff8b3d07c1c27 ensured that redirects would be automatically followed.  As you noticed, though, these changes have yet to be released to the stable release that bootstrap.saltstack.org provides, so a temporary fix is required for kube deployment to work.

Regarding your proposal, I suggest adding a comment before each BS_CURL_ARGS declaration indicating the temporary nature of the fix so that kube doesn't end up carrying it forever.  Is there a reason to supply anything other than -L though?

 - the __fetch_url method in the bootstrap (https://github.com/saltstack/salt-bootstrap/blob/stable/bootstrap-salt.sh#L585) already supplies -s .
 - Why is -k (insecure download) desirable?  I can invoke ``curl -L http://copr.fedoraproject.org/coprs/saltstack/salt/repo/fedora-21/saltstack-salt-fedora-21.repo`` without any issues related to certificate security.
 - I can see how -S might be useful for debugging, but not sure why it should be the default for kube's invocation.  If this needs to be changed, might it make more sense to apply it upstream to the bootstrap?

marun commented:
@arulrajnet I was wondering why I hadn't run into this problem despite having done some kube deployment testing recently.  The recent switch to Fedora23 renders the repo file unnecessary (see #19841), so it doesn't matter that the redirect doesn't work in the current version of the salt bootstrap.  Unless you can replicate the problem against kube master, I suggest closing this PR unmerged.

arulrajnet commented:
@marun Addressed your comments and made the changes. 

I am not yet tested in master.  This issue there in release branch. Peoples first choice is vagrant to hands on with kubernetes. They are blocked because of this issue.

marun commented:
LGTM

marun commented:
@derekwaynecarr This looks good to me, but I don't have merge rights.  Can you please take a look?

marun commented:
@arulrajnet I'm not entirely sure of the policy around kube PR's, but consider collapsing the commits for a cleaner history.

arulrajnet commented:
@marun Squashed commits.

derekwaynecarr commented:
LGTM, thx

arulrajnet commented:
@derekwaynecarr I don't have access to see the failed test logs.

marun commented:
@arulrajnet There is no way this PR could fail the build because vagrant deployment is not tested by e2e.  The build log posted by the bot indicates a test setup failure (Quota 'STATIC_ADDRESSES' exceeded.)

@derekwaynecarr Can you please label as e2e-not-required?

derekwaynecarr commented:
@marun - label updated.  shame the tests are not as stable that it bit us.
